src/org/sipdroid/sipua/SipdroidEngine.java

```diff
@@ -419,6 +427,22 @@ public void unregister(int i) {
 			} else
 				Receiver.onText(Receiver.REGISTER_NOTIFICATION+i, null, 0, 0);
 	}
+
+	public void publishAll () {
+		for (PublishAgent pa : pas) {
+			if (pa != null) {

```

Comment:
@saghul I noticed similar checks on the content of ```ras[]```.  Maybe if presence is disabled for a particular line?